subscriptions/client/database/subscriptions.go

```diff
@@ -0,0 +1,47 @@
+package database
+
+import (
+	"fmt"
+	"main/models"
+)
+
+// GetSubscriptions gets a list of all subscriptions from the database
+func (c *Client) GetSubscriptions() (subscriptions []models.UserSubscription, err error) {
+	rows, err := c.DB.Query("SELECT user_id, subscription_id, status, charge_amount FROM subscriptions")
+	if err != nil {
+		return nil, err
+	}
+	defer rows.Close()
+
+	for rows.Next() {
+		subscription := models.UserSubscription{}
+		if err := rows.Scan(&subscription.UserID, &subscription.SubscriptionID, &subscription.Status, &subscription.ChargeAmount); err != nil {
+			return nil, err
+		}
+		subscriptions = append(subscriptions, subscription)
+	}
+	return subscriptions, nil
+}
+
+// GetUserSubscriptions gets a list of a user's subscriptions from the database
+func (c *Client) GetUserSubscriptions(userID string) (subscriptions []models.UserSubscription, err error) {
+	sqlStatement := "SELECT user_id, subscription_id, status, charge_amount FROM subscriptions WHERE user_id = $1"
+
+	rows, err := c.DB.Query(sqlStatement, userID)
+	if err != nil {
+		return nil, err
+	}
+	defer rows.Close()
+
+	for rows.Next() {
+		subscription := models.UserSubscription{}
+		if err := rows.Scan(&subscription.UserID, &subscription.SubscriptionID, &subscription.Status, &subscription.ChargeAmount); err != nil {
+			return nil, err
+		}
+		subscriptions = append(subscriptions, subscription)
+	}
+	if subscriptions == nil {
+		return subscriptions, fmt.Errorf("no user with id: %s", userID)
```

Comment:
usually it is better to save that kind of data in to logger context rather in to the error itself .

Reply:
also, when query returns a collection it is not necessary to raise not found, empty collection would mean not found as well.